### PR TITLE
Xand y cdeline

### PR DESCRIPTION
**Pull request by cdeline**
I'm getting ready to merge this stuff into development.

At the moment, we have a lot going on in this branch:
-HPC options
-Fix <0 degree elevation error in gendaylit
-axis_tilt
-model chains
-import model parameters from file
-inclined sensors

### CODE REVIEW

**Comment by cdeline** [comment]
Here's what happened with the elevation < 0 errors.  I think we have two options: 

1. assume that our PVLib solar position is correct, and only populate our trackerdict when ghi >0 and solar elevation > 0

2. assume that the weather file is correct, and interpret NaN's as zero tilt, sun just above the horizon. We can take this a step further and calculate what the assumed solar elevation is based on the equation  GHI = DHI + DNI*cos(zenith).

**Comment by cdeline** [comment]
I'm taking option #1 here - ignore any TMY points where pvlib says the sun is not up yet.  There's still a check inside gendaylit which re-creates elevation from dni ghi and dhi if you pass a negative value.  But set1axis should be filtering these values out before it hits gendaylit.

**Comment by shirubana** [comment]
I hadn't seen there was an open pull request. are we ready to merge?

**Comment by cdeline** [comment]
I think this needs to be more finalized before it can go into development.  We should at least have better pytest coverage for the new functions, and get the loading and moduleDict stuff in better shape.

Chris Deline
National Renewable Energy Laboratory
(303) 384-6359

**Comment by cdeline** [comment]
Which development journals do you want to leave in the /docs/ folder?  Here's what I'm thinking of keeping:
(development) Axis_Tilt for Trackers .ipynb
(development) HPC Eval.ipynb
(development) Multiple SceneObjects .ipynb
bifacial_radiance examples.ipynb
1Axis_tracking_example.ipynb
Carport (Mesa) Example, with irradiance mapping of the modules.ipynb
Jan2019 New functionalities examples.ipynb

**Comment by cdeline** [comment]
Yeah, I re-ran some of the notebooks and it gives the deprecation warning.

**Comment by shirubana** [comment]
might be too late but the carport/mesa example would be useful too I think. I haven't seen how outdated it is though. I just have gotten a couple emails regaridng how to do those.

**Comment by shirubana** [comment]
maybe just trim it to one example and comment it more. I can do that later.

**Comment by cdeline** [comment]
Ok. Go ahead and update the carport example in Development directly. I think I left one in.

File: bifacial_radiance/bifacial_radiance.py [hunk 1]
```diff
@@ -2462,249 +2794,446 @@ def saveResults(self, data, reardata=None, savefile=None):
...
-            
-        #if modwanted > 0:
-        #   modwanted = modwanted - 1
-        
-        #if rowwanted > 0:
-        #    rowwanted = rowwanted - 1
-            
-        
+
         if abs(np.tan(azimuth*dtor) ) <=1 or abs(np.tan(azimuth*dtor) ) > 1:
```
Comment (cdeline) [inline bifacial_radiance/bifacial_radiance.py]: What is this line here doing?  isn't tan(azimuth) always either going to be <=1 or >1?

Comment (cdeline) [reply bifacial_radiance/bifacial_radiance.py]: I've removed it - must have been left over from the previous revision.

**Comment by cdeline** [comment]
Next steps:
- branch for GUI, model chain and input file loading

File: bifacial_radiance/bifacial_radiance.py (hunk 1 above)
Comment (shirubana) [reply bifacial_radiance/bifacial_radiance.py]: I think this is from many, mano moons before.

**Comment by shirubana** [comment]
can we delete branch?

**Pull request merged**